src/vs/workbench/parts/debug/node/debugConfigurationManager.ts

```diff
@@ -287,12 +287,16 @@ export class ConfigurationManager implements debug.IConfigurationManager {
 				return null;
 			}
 
-			return this.massageInitialConfigurations(adapter).then(() =>
-				JSON.stringify({
-					version: '0.2.0',
-					configurations: adapter.initialConfigurations ? adapter.initialConfigurations : []
-				}, null, '\t')
-			);
+			return this.massageInitialConfigurations(adapter).then(() => {
+				let editorConfig = this.configurationService.getConfiguration<any>();
+				return JSON.stringify(
+					{
+						version: '0.2.0',
+						configurations: adapter.initialConfigurations ? adapter.initialConfigurations : []
+					},
+					null,
+					editorConfig.insertSpaces ? strings.repeat(' ', editorConfig.tabSize) : '\t');
```

Comment:
I actually had `this.configurationService.getConfiguration('editor')` but TS was not happy with it. Or linter. So I started to try different options, and I think that I tested that without it also worked. My bad.